blogposts/2020/code-search-turned-code-checker.md

```diff
@@ -0,0 +1,296 @@
+---
+title: Code search turned static code checker
+author: Rijnard van Tonder
+authorUrl: https://twitter.com/rvtond
+publishDate: 2020-09-30T14:00-07:00
+tags: [blog]
+slug: code-search-turned-code-checker
+heroImage: /blog/XXX.png
+published: true
+---
+
+<style>
+  .gatsby-highlight {
+    max-width: 100%;
+    width: 40rem;
+    margin-left: auto;
+    margin-right: auto;
+  }
+  table {
+    width: 40rem;
+    border: none;
+  }
+    table th {
+    border: none;
+  }
+  table td {
+    border: none;
+  }
+  table td:nth-child(2n) {
+      text-align: right;
+  }
+    table th:nth-child(2n) {
+      text-align: right;
+  }
+  table tr:nth-child(2n) {
+    background-color: transparent;
+  }
+</style>
+
+I find static code checkers most valuable when they help teach me better ways to
+code in a language or framework. For example, the Go
+[staticcheck](https://staticcheck.io/docs/checks#SA6005) tool finds expensive
+string comparisons like:
+
+```go
+if strings.ToLower(string1) == strings.ToLower(string2) {
+  ...
+}
+```
+
+and suggests instead:
+
+```go
+if strings.EqualFold(string1, string2) {
+  ...
+}
+```
+
+I find these short-and-sweet replacements are a great way to learn framework
```

Comment:
**Nit**: avoid the repetition of leading with "I find" in the first two paragraphs (it doesn't add much here).

```suggestion
These short-and-sweet replacements are a great way to learn framework
```